### ISSUE TITLE
Day and Time Based Exceptions

### ISSUE THREAD

**Issue by tedmau5e** [uamanager/homebridge-calendar-scheduler#285]
I'm not sure if this is possible, or if I'm just underthinking/overthinking how to do this, but I'm trying to figure out how to get this configured such that a HomePod will play at a certain time everyday, except on certain days when it won't play at all (and an event is set for that day to trigger that). Similarly, I would like it to play a certain playlist depending on the date/date range (so I guess an event on the first day of the date range to trigger the change, and then another event to stop it and go back to the regular playlist).

I set up automations in the Home app to set the automations for play and pause at a certain time on the days, but I can't figure out how I would tell it to NOT play if it was a certain day/event on the calendar. I would also like the automations to change the time of play and pause based on a calendar event date range.

Is this possible? Or is there a better way of doing this?

**Comment by uamanager** [comment]
The idea of this plugin is to remove need of scheduling anything on Home app side. Home app scheduling is not flexible and in case of temporary change in schedule, whole new scenario should be created and so on.

That's why I've decided to develop this plugin.
For each trigger of shortcut or automation I'm creating event or series of events in calendar, then configuring plugin to listen for them and trigger respective automation.

Right now morning lights which are simulating sunrise are configured in this way on my side. This allows me to change "sunrise" time for one day from calendar without need to actually disable scenario or create new one.

As for you request, I would organise it in this way:
- create calendar for automation
- schedule calendar events in your desired schedule with appropriate event start date (music start) and end date (music pause)
- configure plugin to listen this calendar and these events
- check in Home app created accessories
- create scenario to start music for open state of accessory
- create scenario to pause music for closed state of accessory

And that's it.
When schedule should be changed - calendar events should be moved on desired time, no need to change anything on Home app side.

Let me know if anything is not clear.

Stay safe 🇺🇦

**Comment by tedmau5e** [comment]
Do start and end events need to be separate events? Or is it possible to have an event last a certain amount of time and the end closes the accessory?

Correct me if I have this right: I would create an automation that detects the sensor for the event opening, which would then trigger the HomePod to play. And then I guess I would have another sensor that waits for the event end to trigger and pause the HomePod, right?

**Comment by uamanager** [comment]
The event accessory is a Contact Sensor. This sensor will be set to the state `Open` once the event starts and to `Close` when the event ends. 

You are able to schedule the event to start at the time when you want the music to start playing and end at the time when the music should stop playing. 

Both automations are directed at one accessory, with the only difference being that the first should listen for the `Open` state and the second for the `Closed` state of the same sensor.

**Comment by uamanager** [comment]
Hey, how is it going? Were you able to setup your automation?

**Comment by uamanager** [comment]
Hey, Finally found time to test this.
I've tried to add `Test` accessory and check if automations still working fine with it.
So I've updated config for calendar event:
```json
{
  "eventName": "Test",
  "eventTriggerOnUpdates": false,
  "caseInsensitiveEventsMatching": false,
  "unsafeEventNames": false
}
```

Then created two automations for this accessory:

1) For Opened state of accessory (when event starts)
![image](https://github.com/uamanager/homebridge-calendar-scheduler/assets/2138444/b068f155-5af9-4f65-a122-982281d4619b)
![image](https://github.com/uamanager/homebridge-calendar-scheduler/assets/2138444/668cb81a-6f2d-4509-b6b7-1da144845986)

2) For Closed state of accessory (when event ends)
![image](https://github.com/uamanager/homebridge-calendar-scheduler/assets/2138444/40f845ff-1796-4458-82a1-49d33455ecfc)
![image](https://github.com/uamanager/homebridge-calendar-scheduler/assets/2138444/99b9c5b1-c281-4b4d-8491-807dd23ad35b)

Using this setup, first scene was enabled when event started and second one enabled when event ended.

Let me know if your automation configuration differs from my examples.

**Comment by tedmau5e** [comment]
Hey!

Thanks, I ended up getting it figured out and working correctly not long after my last post/update. And then something happened with networking or Windows that caused a complete failure and I had to reset everything back up (using Ubuntu instead of Windows) and reset the HomePod because of the networking issue, but now everything is working perfectly!

Thank you for not forgetting and checking back in!
